### ISSUE TITLE
Improved `--try-toolchain` overly picky

### ISSUE THREAD

**Issue by bartoldeman** [easybuilders/easybuild-framework#2592]
The improved `--try-toolchain` now rejects cases that used to work perfectly, where the toolchain can be "downgraded" to a subtoolchain.

Example:
```
eb libxc-4.2.3-intel-2018a.eb --try-toolchain=iccifort,2018.3.222-GCC-7.3.0-2.30
== temporary log file in case of crash /localscratch/eb-oe4GFb/easybuild-77BHp6.log
ERROR: No possible mapping from source toolchain spec {'lapack_family': None, 'name': 'iimpi', 'blas_family': None, 'version': '2018a', 'cuda': None, 'comp_family': 'Intel', 'mpi_family': 'IntelMPI'} to target toolchain hierarchy specs [{'lapack_family': None, 'name': 'GCCcore', 'blas_family': None, 'version': '7.3.0', 'cuda': None, 'comp_family': 'GCC', 'mpi_family': None}, {'lapack_family': None, 'name': 'iccifort', 'blas_family': None, 'version': '2018.3.222-GCC-7.3.0-2.30', 'cuda': None, 'comp_family': 'Intel', 'mpi_family': None}]
```

@ocaisa

**Comment by ocaisa** [comment]
Hmm, so I'm on the fence with this one. I think that if you really want to do that you should add a `--force` option and we could revert it back to regex method in that case.

The issue here for me would be that you should be only doing that if you know what you are doing, so protecting against this is reasonable default behaviour.

**Comment by boegel** [comment]
+1 on @ocaisa view on this, you should take care when mapping from a full toolchain to a lower subtoolchain that has less capabilities.

So, there should be a hint in the error message that mentions that you can use `--force` if desired, and then also a clear warning when `--force` is used.

The error message could also be improved a bit, e.g. `ERROR: Toolchain iccifort is not equivalent to intel toolchain in terms of capabilities: ...`?

**Comment by bartoldeman** [comment]
+1 sure I don't mind force. Issue is here that there quite a few easyconfigs compiled with foss or intel that don't use MKL/OpenBLAS or MPI or both; as we try to compile things minimally we often lower the toolchain.

**Comment by ocaisa** [comment]
If you need a workaround for now you can use
```
--try-amend=''
```
since when `--try-toolchain` is coupled with another try option it will revert to using the regex method.

**Comment by ocaisa** [comment]
I think we will soon be at a place where we can be stricter on the toolchain used for easyconfigs and put them at the right level from the outset. We would just need to support mapping up to the top level, and that should be do-able fairly easily now.

**Comment by boegel** [comment]
fixed in #2605
